app/ante_gate.go

```diff
@@ -41,3 +54,42 @@ func (ad AnteSpamMigitationDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, sim
 
 	return next(ctx, tx, simulate)
 }
+
+// AnteDecorator
+func (ad AnteRestrictUbedrockDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate bool, next sdk.AnteHandler) (sdk.Context, error) {
+	fmt.Printf("[LOG] AnteRestrictUbedrockDecorator - AnteHandle - LINE 59\n")
+	fmt.Printf("[LOG] AnteRestrictUbedrockDecorator - AnteHandle - LINE 62\n")
+	sigTx, ok := tx.(authsigning.SigVerifiableTx)
+	if !ok {
+		return ctx, sdkerrors.Wrap(sdkerrors.ErrTxDecode, "invalid transaction type")
+	}
+	sigTxSignatures, err := sigTx.GetSignaturesV2()
+	if err != nil {
+		return ctx, sdkerrors.Wrap(err, "getting signatures of tx is failed")
+	}
+
+	messages := sigTx.GetMsgs()
+	fmt.Printf("[LOG] AnteRestrictUbedrockDecorator - AnteHandle - messages : %v\n", messages)
+
+	msgCreateAccount, ok := messages[0].(*types.MsgCreateAccount)
```

Comment:
MsgSend:
1. check if denom bedrock is in Amount. https://github.com/cosmos/cosmos-sdk/blob/main/x/bank/types/tx.pb.go#L39
2. if true, check if receiver is in KYC address.

Reply:
Fix in https://github.com/Pylons-tech/pylons/pull/1018/commits/08f2e591fc712d365c428fa741c5d036f8f4b528